src/oldreader_api.cpp

```diff
@@ -65,28 +66,24 @@ std::string oldreader_api::retrieve_auth() {
 	}
 
 	std::string pass = cfg->get_configvalue("oldreader-password");
+    std::string pass_file = cfg->get_configvalue("oldreader-passwordfile");
+    std::string pass_eval = cfg->get_configvalue("oldreader-passwordeval");
 	if ( pass == "" ) {
-		wordexp_t exp;
-		std::ifstream ifs;
-		wordexp(cfg->get_configvalue("oldreader-passwordfile").c_str(),&exp,0);
-		if (exp.we_wordc > 0) {
-			ifs.open(exp.we_wordv[0]);
-		}
-		wordfree(&exp);
-		if (!ifs.is_open()) {
-			if (!flushed) {
-				std::cout << std::endl;
-				std::cout.flush();
-			}
-			// Find a way to do this in C++ by removing cin echoing.
-			pass = std::string( getpass("Password for The Old Reader: ") );
-		} else {
-			std::getline(ifs, pass);
-			if (pass == "") {
-				return "";
-			}
-		}
+        if ( pass_file != "" ) {
+            pass = read_password(pass_file.c_str());
+        } else if ( pass_eval != "") {
+            pass = eval_password(pass_eval.c_str());
+        }
 	}
+
+    if ( pass == "" ) {
+        if (!flushed) {
+            std::cout << std::endl;
+            std::cout.flush();
+        }
+        // Find a way to do this in C++ by removing cin echoing.
+        pass = std::string( getpass("Password for The Old Reader: ") );
+    }
```

Comment:
Original version of the function returned "" if no password was obtained. Please do the same here.